### PR TITLE
GIX-1656: Apply same rules and show errors when creating and renaming canisters

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

The rules when naming canisters on creating or afterwards was not consistent and there was no feedback to the user about them

In this PR: I apply the same rules in both flows and show an error message to the user when needed.

# Changes

* Add `errorMessage` parameter to `TextInputForm`.
* New canister util `errorCanisterNameMessage` that returns error or `undefined`.
* Use new parameter with new util in `RenameCanisterModal`.
* Use new parameter with new util in `CreateCanisterModal`.
* Change the rules to disable the button in `RenameCanisterModal`.

# Tests

* Test new parameter in `TextInputForm`.
* Test new cansiter util `errorCanisterNameMessage`.
* Add a test case in `CreateCanisterModal`.
* Add a test case in `RenameCanisterModal` and change the expectation when input is empty.

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

**Comment by lmuntaner** [comment]
New UX:

Allow renaming to empty string:
https://github.com/dfinity/nns-dapp/assets/4550653/32a37ec0-2113-4d07-82a8-96d8109907ee

Do not allow renaming to long names:
https://github.com/dfinity/nns-dapp/assets/4550653/a059f3fd-83ac-4fc1-96f2-750e437c4f7a

Do not allow creating canisters with long names.
https://github.com/dfinity/nns-dapp/assets/4550653/51434e2e-4433-45c5-8250-c15e79eda942

**Comment by lmuntaner** [comment]
@dskloetd please review

File: frontend/src/lib/modals/canisters/RenameCanisterModal.svelte [hunk 1]
```diff
@@ -57,9 +59,11 @@
     bind:text={currentName}
     placeholderLabelKey="canister_detail.rename_canister_placeholder"
     busy={$busy}
-    disabledConfirm={isNullish(currentName) ||
-      currentName?.length === 0 ||
-      $busy}
+    disabledConfirm={$busy ||
+      (nonNullish(currentName) &&
+        currentName?.length >= MAX_CANISTER_NAME_LENGTH)}
```
Comment (dskloetd) [inline frontend/src/lib/modals/canisters/RenameCanisterModal.svelte:64]: Shouldn't it be `>` instead of `>=`?
A length equal to the max length should be allowed, right?

Comment (dskloetd) [inline frontend/src/lib/modals/canisters/RenameCanisterModal.svelte:64]: After checking `nonNullish(currentName)`, you don't need the `?` in `currentName?.length`, right?

But also, `null > 24` evaluates to `false` so I guess you don't need the `nonNullish(currentName)` check unless TypeScript requires it?

File: frontend/src/lib/utils/canisters.utils.ts [hunk 2]
```diff
@@ -75,3 +78,14 @@ export const canisterStatusToText = (status: CanisterStatus): string => {
 
 export const mapCanisterId = (canisterId: CanisterId | string): CanisterId =>
   typeof canisterId === "string" ? Principal.fromText(canisterId) : canisterId;
+
+export const errorCanisterNameMessage = (name: string | undefined) => {
+  if (nonNullish(name) && name?.length >= MAX_CANISTER_NAME_LENGTH) {
```
Comment (dskloetd) [inline frontend/src/lib/utils/canisters.utils.ts:83]: `>` instead of `>=`?

File: frontend/src/tests/lib/modals/canisters/CreateCanisterModal.spec.ts [hunk 3]
```diff
@@ -161,6 +164,37 @@ describe("CreateCanisterModal", () => {
...
+    await waitFor(() => expect(getIcpToCyclesExchangeRate).toBeCalled());
+    // wait to update local variable with conversion rate
+    await tick();
+
+    const accountCards = queryAllByTestId("account-card");
+    expect(accountCards.length).toBe(2);
+
+    fireEvent.click(accountCards[0]);
+
+    // Enter Name Screen
```
Comment (dskloetd) [inline frontend/src/tests/lib/modals/canisters/CreateCanisterModal.spec.ts:181]: Up to here it's all boiler plate test setup, right?
If this is shared between multiple tests, maybe it can we a funfction?

File: frontend/src/tests/lib/modals/canisters/CreateCanisterModal.spec.ts [hunk 4]
```diff
@@ -161,6 +164,37 @@ describe("CreateCanisterModal", () => {
...
+    await waitFor(() =>
+      expect(queryByTestId("create-canister-name-form")).toBeInTheDocument()
+    );
+
+    const longName = "a".repeat(MAX_CANISTER_NAME_LENGTH + 1);
```
Comment (dskloetd) [inline frontend/src/tests/lib/modals/canisters/CreateCanisterModal.spec.ts:186]: Can you also add a test that `"a".repeat(MAX_CANISTER_NAME_LENGTH)` does not cause an error?

File: frontend/src/lib/modals/canisters/RenameCanisterModal.svelte (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/modals/canisters/RenameCanisterModal.svelte:64]: True, it should be `>`

Comment (lmuntaner) [reply frontend/src/lib/modals/canisters/RenameCanisterModal.svelte:64]: Discussed on slack. TS requres the check. but `?` is not necessary.

File: frontend/src/lib/utils/canisters.utils.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/lib/utils/canisters.utils.ts:83]: correct

File: frontend/src/tests/lib/modals/canisters/CreateCanisterModal.spec.ts (hunk 4 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/modals/canisters/CreateCanisterModal.spec.ts:186]: Ok, I'll add it case also in the util test

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

**Review by dskloetd — APPROVED** [review]
